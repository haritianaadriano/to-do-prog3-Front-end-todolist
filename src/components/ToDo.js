import Task from "./Task"

export default function ToDo({toDoList, toDone}){
    return(
        <div>
            <h3>To-do</h3>
            <ul
                data-testid="todo-id"
            >
                {toDoList.map((todo, index) => (
                    <Task 
                        todo={todo} 
                        index={index} 
                        toDone={toDone}
                    />
                ))}
            </ul>
        </div>
    )
}