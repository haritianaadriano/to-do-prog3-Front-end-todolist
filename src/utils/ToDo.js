export default function ToDo(props){
    let [toDoList, toDone] = props;
    return(
        <>
            <h3>To-do</h3>
            <ul>
                {toDoList.map((todo, index) => (
                <li key={index}>
                    {todo}
                    <button 
                        type="button" 
                        onClick={() => toDone(index)}
                    >
                        Done
                    </button>
                </li>
                ))}
            </ul>
        </>
    )
}