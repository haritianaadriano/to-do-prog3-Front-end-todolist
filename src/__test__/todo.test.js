import {render, cleanup, screen} from "@testing-library/react";
import ToDo from "../components/ToDo";

test("should_render_todo_component()", ()=>{
    render(<ToDo/>);
    const todoElement = screen.getByTestId("todo-id");
    expect(todoElement).toBeInTheDocument();
})