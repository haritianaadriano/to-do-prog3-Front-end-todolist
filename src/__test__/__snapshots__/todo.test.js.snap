// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo [snapshot] renders correctly 1`] = `
<div>
  <h3>
    To-do
  </h3>
  <ul
    data-testid="todo-id"
  >
    <div>
      <li
        data-testid="task-id"
      >
        ohatra__1
        <div>
          <input
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </li>
    </div>
    <div>
      <li
        data-testid="task-id"
      >
        ohatra__2
        <div>
          <input
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </li>
    </div>
  </ul>
</div>
`;
